### DISCUSSION TITLE
Meeting showing as temporarily inactive when it is active

### DISCUSSION THREAD

**Discussion by cawood402** [code4recovery/12-step-meeting-list#531] [Q&A] [answered]
https://aachilternthames.org.uk/meetings/parmoor-meditation-meeting/

I believe the issue is related to the address which is being interpreted as approximate.

The address is 
Frieth Village Hall, Frieth, Henley-on-Thames RG9 6PR, UK

**Comment by tech2serve** [comment]
Hi @cawood402,

I geocoded the address you have above and it came back from Google to me as non-approximate. 

The flag representing whether or not a meeting is temporarily inactive or not is set primarily from how the admin checks the block about whether or not a meeting can be attended in person (assuming you’re using the latest version of TSML).

If you’ve checked “yes” to being able to attend in person, the meeting should not show up as inactive. So the first thing I suggest is to double-check that you have “yes” selected for being able to attend in person.

If that is not the issue, then we need to try and determine if your address is cached with the incorrect approximate response. To do this, you might be able to open a developer console and then re-enter the address on edit meeting page. Watch the output in the console and you should see a `Geocoded`. Click on the arrow next to `Geocoded` and you should see details of the response. Please screen cap that and post it here as a comment to this answer.

**Reply by cawood402** [reply]
Hi

Thanks for the quick response. I've uploaded a screen shot of the geocode result which does say it is cached. But also says approximate = 'no'.

![Screenshot 2021-09-23 at 10 21 00](https://user-images.githubusercontent.com/76901455/134483258-a595a280-2578-4d38-96c1-3b32215f5c85.png).

I can confirm that I have checked the 'Can I currently attend this meeting in person?' meeting. After I've saved the value goes back to 'No' and the meeting remains temporarily inactive.

**Reply by tech2serve** [reply]
Ok, this is odd. Are you seeing any errors show in the console along with `Geocoded`?

@brianw-area24, any ideas? Might be a bug, no?

**Reply by cawood402** [reply]
Hi
Yes, there are a few errors in the console
![Screenshot 2021-09-25 at 14 32 33](https://user-images.githubusercontent.com/76901455/134773347-965484ff-be6e-4ec5-ac5a-2e336d3d86a5.png)

I can edit and save other fields on the page. I have also successfully set the meeting in person to Yes and saved with another valid address. However, when I set it back to this address the meeting in person flag returns to 'no'. The errors occur in both cases.

**Reply by tech2serve** [reply]
Very weird. 

At this time, I still cannot confirm if this is a TSML bug; if something got weird in your cache; or if something else in your site is messing with this particular entry.

My suggestion is that we should try an edit or delete your cache. For that, I’d ask you to send me an email to devs@code4recovery.org and I’ll send you directions on how to do this. This does assume you have access to your database?

If you’re willing, please download and install [our debugging tool](https://github.com/code4recovery/tsml_debug) which will let me view more data on your site. You’ll have to install it manually.

In the meantime, you can try a couple of things. What version of Wordpress do you have installed? They’ve been doing a lot of migration work with jQuery and for some versions, a jQuery Migrate plugin is necessary. However, with newer versions, it probably isn’t required or may even interfere with other processes. I’m very curious about that jQuery TypeError. Do you have a plugin or something involving seven days? Anyway, try disabling plugins, update Wordpress, etc., to see if the issue resolves. Likewise with the other error related to your cookie consent. Issues caused by other errors don’t usually affect just a single meeting, but it might help site performance at the very least to resolve those errors.

**Comment by tech2serve** [comment, marked as answer]
Hey @cawood402,

With tremendous gratitude for @joshreisner, we believe we’ve figured out what is going on. I won’t bore you with all the details, but we suspect the way Google is treating the address is to have it as approximate in one form and non-approximate in another form. This is confusing TSML and your site. Our immediate solution is to have you add some code in your site to override Google’s address response.

Here is the basic reference for how to [override Google’s geocoding response](https://github.com/code4recovery/12-step-meeting-list/wiki/Frequently-Asked-Questions#i-need-to-correct-a-meeting-address-or-change-a-pins-location) in TSML.

Hopefully you have a child theme set up, but if not you’ll probably need to do that or find a plugin that allows you to customize plugin functions. 

In your custom `functions.php`, add:

```
if (function_exists('tsml_custom_types')) {
	tsml_custom_addresses(array(
        	'Frieth Village Hall, Frieth, Henley-on-Thames RG9 6PR, UK' => array(
	        	'formatted_address' => 'Frieth Village Hall, Frieth, Henley-on-Thames RG9 6PR, UK',
		        'city' => 'Frieth',
		        'latitude' => 51.6051635,
		        'longitude' => -0.852636,
		        'approximate' => 'no'
	        ),
	));	
}
```

Please let me know if this makes sense, and I’m happy to try and answer any questions.

**Reply by cawood402** [reply]
Brilliant!!!

Thank you so much for your time and expertise. 
I copied the code in to the functions.php of the child theme and then updated the address with the 'in person' set to yes and this time it saved and the meeting displays as required.

Many thanks again.

**Reply by cawood402** [reply]
The address and meeting status are now correct. But I'm sorry to have to report that the map does not appear on the meetings directory page: https://aachilternthames.org.uk/meetings/parmoor-meditation-meeting/

Is this related to the change? 

There are errors in the console:
![Screenshot 2021-09-26 at 08 35 51](https://user-images.githubusercontent.com/76901455/134798470-a6a79045-48ca-401c-84f1-55797067e9ed.png)

**Reply by tech2serve** [reply]
Dang. Given that this mapping issue only seems to be present with the one meeting, we need to assume they're related. I'm still not clear if we have an actual bug (well, leaning towards yes as it relates to the broader problem in how we handle Google's response) or if this is something we can fix by tweaking your cache. But there is another option, and for that I want to try and duplicate your issue on my site.

What versions of Wordpress are you running? And if you know the PHP version?

Are you willing to download the debugging tool I mentioned above? I'd like to try and look at your cache.

**Reply by tech2serve** [reply]
Hey @cawood402,

There is a minor mistake in the code I gave you, but I do not think the correction will change things. But, let's give it a try to be sure?

```
if (function_exists('tsml_custom_addresses')) {
    tsml_custom_addresses(array(
        'Frieth Village Hall, Frieth, Henley-on-Thames RG9 6PR, UK' => array(
            'formatted_address' => 'Frieth Village Hall, Frieth, Henley-on-Thames RG9 6PR, UK',
                'city' => 'Frieth',
                'latitude' => 51.6051635,
                'longitude' => -0.852636,
                'approximate' => 'no'
	    ),
    ));	
}
```

Only the first line, where we test for the existence of the function (i.e., the plugin is installed), changes.

**Reply by tech2serve** [reply]
So we believe the reason the map isn't working is that something is erasing the latitude and longitude values, and we've not been able to duplicate this issue on two of our development sites. 

Although I'm going to keep looking at a few things, I believe you need to figure out the cause of your any site errors. These are likely coming from another plugin or something with your base configuration; I do not believe they are related to TSML. Usually the best way to test these is to disable all other plugins and see if the problem resolves. Then add the plugins back in one at a time to see what creates the problem. You also should look at any other customizations you have made (except the address override above).

Please keep us posted. Handling addresses outside the US is one of our trickier challenges, and we'd like to improve this feature for you all.

**Reply by tech2serve** [reply]
Hey @cawood402, I wanted to follow up with you and see if you had any luck with this by disabling the other plugins?

**Comment by jmh2911** [comment]
I noticed using the Support Groups type option that the "Open" box was missing in the meeting information section. When I changed the group type to AA, the "Open" box was available. It has to be checked to show the group as active, then you can switch the group type back to whatever.

**Reply by tech2serve** [reply]
Hello @jmh2911, Did you mean for this to be an "answer" for @cawood402? Or are you having an issue you need help with. If the latter, please open a new question. Thanks.
